src/cache/azure.rs

```diff
@@ -70,18 +84,27 @@ impl Storage for AzureBlobCache {
     async fn put(&self, key: &str, entry: CacheWrite) -> Result<Duration> {
         let start = Instant::now();
         let data = entry.finish()?;
+        let key = self.normalize_key(key);
 
         let _ = self
             .container
-            .put(key, data, &self.credentials)
+            .put(&key, data, &self.credentials)
             .await
             .context("Failed to put cache entry in Azure")?;
 
         Ok(start.elapsed())
     }
 
     fn location(&self) -> String {
-        format!("Azure, container: {}", self.container)
+        format!(
+            "Azure, container: {}, {}",
```

Comment:
Let's match the format used by GCS.
```suggestion
            "Azure, container: {}, key_prefix: {}",
```